src/marqo/tensor_search/models/add_docs_objects.py

```diff
@@ -61,14 +62,25 @@ class Config:
     index_name: str
     auto_refresh: bool
     device: Optional[str]
-    non_tensor_fields: List = Field(default_factory=list)
+    non_tensor_fields: Optional[List] = Field(default_factory=list)
+    tensor_fields: Optional[List] = Field(default_factory=None)
     image_download_thread_count: int = 20
     image_download_headers: dict = Field(default_factory=dict)
     use_existing_tensors: bool = False
     mappings: Optional[dict] = None
     model_auth: Optional[ModelAuth] = None
 
-    def __init__(self, **data: dict):
+    @root_validator
+    def validate_fields(cls, values):
```

Comment:
We are moving towards the primary validation being included in the AddDocsParams object (so raising user errors  rather than internal errors here)

Reply:
As discussed this will be refactored as a separate PR and will remain unchanged for this one